server/pulp/server/async/tasks.py

```diff
@@ -271,7 +271,8 @@ def apply_async(self, *args, **kwargs):
         ### This is a temporary workaround for https://github.com/celery/celery/issues/2216 ###
         queue_name_to_save = queue
         if kwargs.get('exchange', None) == DEDICATED_QUEUE_EXCHANGE:
-            kwargs['queue'] = queue.rstrip('.dq')
+            # Strip off '.dq' from the end of the queue name
+            kwargs['queue'] = queue[:-3]
```

Comment:
I assume this means you're 100% confident that the queue value will always end in ``.dq``. If not, you could use something like ``if foo.endswith('.dq'):`` or a regex-based substitution.